nncf/pruning/base_algo.py

```diff
@@ -233,7 +233,7 @@ def pruning_rate_for_filters(minfo: PrunedModuleInfo):
         Calculates sparsity rate for weight filter-wise.
         """
         weight = minfo.module.weight
-        filters_sum = weight.view(weight.size(0), -1).sum(axis=1)
+        filters_sum = weight.view(weight.size(minfo.module.target_weight_dim_for_compression), -1).sum(axis=1)
```

Comment:
Here you should transpose weight (in case of target_weight_dim_for_compression != 0) first.
Instead, you will mix filters by reshaping tensor and calculate incorrect filter-wise pruning rate.

Reply:
fixed